src/libpgmoneta/restore.c

```diff
@@ -1793,6 +1899,82 @@ incremental_rfile_initialize(int server, char* label, char* relative_dir, char*
    return 1;
 }
 
+static int
+split_file_path(char* path, char** relative_path, char** bare_file_name)
```

Comment:
All variables in the beginning of the function, and an `error` label to goto.

Clean up and return should be last the every function